pkg/registry/batch/job/strategy.go

```diff
@@ -211,23 +211,30 @@ func generateSelector(obj *batch.Job) {
 	// The job-name label is unique except in cases that are expected to be
 	// quite uncommon, and is more user friendly than uid.  So, we add it as
 	// a label.
-	_, found := obj.Spec.Template.Labels["job-name"]
-	if found {
-		// User asked us to not automatically generate a selector and labels,
-		// but set a possibly conflicting value.  If there is a conflict,
-		// we will reject in validation.
-	} else {
-		obj.Spec.Template.Labels["job-name"] = string(obj.ObjectMeta.Name)
+	jobNameLabels := []string{"job-name", batch.JobLabelPrefix + "job-name"}
+	for _, value := range jobNameLabels {
+		_, found := obj.Spec.Template.Labels[value]
+		if found {
```

Comment:
Does this comment explain it?

```
	// If selector generation was requested, then expected labels must be
	// present on pod template, and must match job's uid and name.  The
	// generated (not-manual) selectors/labels ensure no overlap with other
	// controllers.  The manual mode allows orphaning, adoption,
	// backward-compatibility, and experimentation with new
	// labeling/selection schemes.  Automatic selector generation should
	// have placed certain labels on the pod, but this could have failed if
	// the user added coflicting labels.  Validate that the expected
	// generated ones are there.
```

I think that validation requires that you must have these labels correspond to Name and ControllerUId so not entirely sure what we gain from this.

Reply:
So let me see if I get this right (not super familair with Job):

If the user wants a manual selector, they say so and then put whatever value they want in the selector, right?  This function appears to only be called if they didn't set the `manualSelector` field.

So the below comment "User asked us to not automatically generate a selector" is wrong - this whole function is only called when the user DID ask us to generate a selector.  Or do I still misunderstand?

Reply:
It should say: "User asked us to automatically generate a selector, but set manual labels, maybe conflicting"

Reply:
I didn't update this comment in this PR so I opened up https://github.com/kubernetes/kubernetes/pull/116215 where we can fix that.

Reply:
It's ok to fix such small things in a PR that is doing something else. But up to you.

Reply:
Good to know! I usually like to follow the approach of small PRs and keeping them focused on a particular thing.  I didn't really follow my own policy in this one so I've been trying to push things unrelated to this PR to another one.  Mostly so reviews can be detailed and easy to get through.  

I also think I want to look a bit more at manual selectors hence https://github.com/kubernetes/kubernetes/issues/116137

Reply:
#116215 was merged so I think we get resolve this.